tests/lib/rules/block-attributes-order.js

```diff
@@ -0,0 +1,345 @@
+            order: ['module', 'src', 'lang']
+          }
+        ]
+      ]
+    },
+    {
+      filename: 'test-attributes-options-not-specify-some-attribute-2.vue',
+      code: '<style module scoped src="./style.css" lang="less"></style>',
+      options: [
+        [
+          {
+            element: 'style',
+            order: ['module', 'src', 'lang']
+          }
+        ]
+      ]
+    },
+    {
+      filename: 'test-attributes-options-not-specify-some-attribute-3.vue',
+      code: '<style scoped module src="./style.css" lang="less"></style>',
+      options: [
+        [
+          {
+            element: 'style',
+            order: ['module', 'src', 'lang']
+          }
+        ]
+      ]
+    },
+    {
+      filename: 'test-attributes-options-with-batch-order-1.vue',
+      code: '<style module scoped src="./style.css" lang="less"></style>',
+      options: [
+        [
+          {
+            element: 'style',
+            order: ['module', ['scoped', 'src'], 'lang']
+          }
+        ]
+      ]
+    },
+    {
+      filename: 'test-attributes-options-with-batch-order-2.vue',
+      code: '<style module src="./style.css" scoped lang="less"></style>',
+      options: [
+        [
+          {
+            element: 'style',
+            order: ['module', ['scoped', 'src'], 'lang']
+          }
+        ]
+      ]
+    },
+    {
+      filename: 'test-custom-block-with-attributes-options.vue',
+      code: '<foobarbaz foo bar baz="baz"><foobarbaz>',
+      options: [
+        [
+          {
+            element: 'foobarbaz',
+            order: ['foo', 'bar', 'baz']
+          }
+        ]
+      ]
+    },
+    {
+      filename: 'test-using-element-selector-1.vue',
+      code: '<script lang="ts" setup functional></script>',
+      options: [
+        [
+          {
+            element: 'script[setup]',
+            order: ['lang', 'setup', 'functional']
+          }
+        ]
+      ]
+    },
+    {
+      filename: 'test-using-element-selector-2.vue',
+      code: '<script functional lang="ts"></script>',
+      options: [
+        [
+          {
+            element: 'script[setup]',
+            order: ['lang', 'setup', 'functional']
+          },
+          {
+            element: 'script:not([setup])',
+            order: ['functional', 'lang']
+          }
+        ]
+      ]
+    },
+    {
+      filename: 'test-using-regex-order-1.vue',
+      code: '<style scoped module lang="less" src="./style.css"></style>',
+      options: [
+        [
+          {
+            element: 'style',
+            order: ['^(?!(src|lang)$).*', 'lang', 'src']
+          }
+        ]
+      ]
+    }
+  ],
+  invalid: [
+    {
+      filename: 'test-template-attributes-with-invalid-order.vue',
+      code: '<template src="./template.html" lang="pug" functional></template>',
+      errors: [
+        {
+          type: 'VAttribute',
+          message: 'Attribute "lang" should go before "src".'
+        },
```

Comment:
Please add `line` and `column` of the reported error to all the expected errors in this file. However, you can also remove the `type: 'VAttribute'` everywhere.